### DISCUSSION TITLE
remark-toc: Rendering header tags in component into TOC

### DISCUSSION THREAD

**Discussion by jacksongoode** [remarkjs/remark#568] [Q&A] [answered]
I feel like the answer must be here somewhere, but I am using Docusaurus and am just curious if it is possible to generate a TOC without markdown, but through rendering html in a React component.

Either this or is there a way to send html as markdown into the interpreter?

**Comment by wooorm** [comment]
This sounds like a question for docusaurus: you‘re asking on remark, which is about markdown.

Anyway, yeah: there are loads of ways to make a toc. From markdown, from HTML, from React. But this is a very broad question.

What is the interpreter? Why send HTML as Markdown? This all sounds like an [xy problem](http://xyproblem.info)

**Comment by jacksongoode** [comment, marked as answer]
https://github.com/facebook/docusaurus/issues/1921#issuecomment-550402762

I should clarify I am just dipping my toes into this topic. I am rendering html with a custom component through .mdx. I was brought to remark-toc through the above discussion as I thought that maybe the functionality I'm describing is apart of remark rather than Docusaurus.

**Reply by wooorm** [reply]
MDX is build on remark and rehype. remark is markdown. rehype is for html. You can use `remark-slug` and `remark-toc` on markdown. there is also a [rehype-toc](https://github.com/JS-DevTools/rehype-toc) if you want to do it in html. Lastly, MDX supports components.
